### PR TITLE
Definition schema class

### PR DESCRIPTION
**Pull request by funkyfuture**
this includes a commit that improves DefinitionSchema and one that fixes an issue that came up in the doctests for the first time.

### CODE REVIEW

**Comment by funkyfuture** [comment]
i added some commits to purge the in-code-backlog

well, more tests would be possible. :-/

File: docs/customize.rst [hunk 1]
```diff
@@ -51,6 +51,9 @@ matters, we can use it to validate all odd values:
     >>> v.validate({'oddity': 9})
     True
 
+In a schema schema you can use space characters instead of underscores, e.g.
+``{'oddity': {'is odd': 42'}}`` is an alias for ``{'oddity': {'is odd': 42'}}``.
```
Comment (nicolaiarocci) [inline docs/customize.rst]: ?

**Comment by nicolaiarocci** [comment]
Nice. Would you also update CHANGES to reflect the change from ValidationError to DocumentError, and error constants which have changed name?

**Comment by funkyfuture** [comment]
update: i added some rules related to normalization. @misja this might be of interest for you.

tomorrow or the day after i'll propably fully update [`CHANGES`](https://github.com/nicolaiarocci/cerberus/pull/135#issuecomment-129334608) and add a `trail`-property to `Validator`.

**Comment by funkyfuture** [comment]
a'ight, i added the last commits to this pr.
the `trail`-property is just added more as proof-of-concept , without much use of it. i will certainly use it when refactoring error-representation (#93). it may be useful for rules like dependencies on parent- / siblings-fields, or conditional validation like requested in #136.

**Pull request merged**